tests/cpp/c_api/test_c_api.cc

```diff
@@ -162,7 +163,7 @@ TEST(CAPI, ConfigIO) {
 TEST(CAPI, JsonModelIO) {
   size_t constexpr kRows = 10;
   size_t constexpr kCols = 10;
-  dmlc::TemporaryDirectory tempdir;
```

Comment:
The current implementation of `dmlc::TemporaryDirectory` is not capable of handling UTF-8 paths. We should deprecate `dmlc::TemporaryDirectory` in favor of the standard function `std::filesystem::temp_directory_path`.